diagnostics/src/status_handlers.cpp

```diff
@@ -0,0 +1,63 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "status_handlers.h"
+
+namespace concord::diagnostics {
+
+void StatusHandlers::registerHandler(StatusHandler handler) {
+  std::lock_guard<std::mutex> guard(mutex_);
+  if (status_handlers_.count(handler.name)) {
+    throw std::invalid_argument(std::string("StatusHandler already exists: ") + handler.name);
+  }
+  status_handlers_.insert({handler.name, handler});
+}
+
+std::string StatusHandlers::get(const std::string& name) const {
+  std::lock_guard<std::mutex> guard(mutex_);
+  if (status_handlers_.count(name)) {
+    return status_handlers_.at(name).f();
+  }
+  return "*--STATUS_NOT_FOUND--*";
```

Comment:
nit: Maybe add this as static const std::string STATUS_NOT_FOUND {"*--STATUS_NOT_FOUND--*"} to avoid copy-pasting the text all around the code.

Reply:
These were all just moved to a different file. I prefer to postpone this change.